src/ast/util.ts

```diff
@@ -3,33 +3,71 @@ import * as A from "./ast";
 import { isLiteral, FactoryAst } from "./ast-helpers";
 import { dummySrcInfo, SrcInfo } from "../grammar";
 
-export const getAstUtil = ({ createNode }: FactoryAst) => {
+export const getAstUtil = ({ createNode, cloneNode }: FactoryAst) => {
+    /**
+     * @deprecated, since it uses dummySrcInfo. Use makeUnaryExpressionLoc instead.
+     *
+     */
     function makeUnaryExpression(
         op: A.AstUnaryOperation,
         operand: A.AstExpression,
-    ): A.AstExpression {
+    ): A.AstOpUnary {
```

Comment:
please revert all the unrelated changes, open a new issue and solve it in a new PR